core/styles/components.css

```diff
@@ -214,3 +214,44 @@
   width: 100%;
   text-align: center;
 }
+
+/* toggle button */
+
+input[type='checkbox'] {
+  position: relative;
+  width: 80px;
+  height: 20px;
+  appearance: none;
+  border-radius: 2rem;
+  cursor: pointer;
+  color: red;
+  border: 2px black solid;
+  padding-right: 4px;
+}
+
+input[type='checkbox']::before {
+  content: '';
+  margin-top: 2.5px;
+  margin-bottom: 2.5px;
+  margin-left: 2.5px;
+  width: 12px;
+  height: 12px;
+  border-radius: 50%;
+  background: red;
+  position: absolute;
+  transition: 0.3s;
+}
+
+input[type='checkbox']:checked::before {
+  transform: translateX(60px);
+}
```

Comment:
Please make the toggle button about half its current width. Also, if possible `margins`, `width`, and `height` should be relative to their parent (use `%`).

Reply:
ok i have changed width
I tried to implement percentage wherever it was possible without altering the design too much